tests: test_patchelf_is_relocatable fails on LLNL/Quartz

The ``test_patchelf_is_relocatable`` test fails on LLNL's Quartz since `patchelf` is a bash script wrapper, not an executable as required by the test.

Perhaps ``is_binary`` should be checking the subtype from ``mime_type`` and declaring binary if it is ``x-shellscript``?

### Steps to reproduce the issue

```console
$ spack test -k test_patchelf_is_relocatable
...
```

### Error Message

```console
_________________________ test_patchelf_is_relocatable _________________________

    @pytest.mark.requires_executables(
        'patchelf', 'strings', 'file'
    )
    def test_patchelf_is_relocatable():
        patchelf = spack.relocate.get_patchelf()
>       assert spack.relocate.is_binary(patchelf)
E       AssertionError: assert False
E        +  where False = <function is_binary at 0x2aaab19c62f0>('/usr/tce/bin/patchelf')
E        +    where <function is_binary at 0x2aaab19c62f0> = <module 'spack.relocate' from '$HOME/spack/clean/spack/lib/spack/spack/relocate.py'>.is_binary
E        +      where <module 'spack.relocate' from '$HOME/spack/clean/spack/lib/spack/spack/relocate.py'> = spack.relocate

relocate.py:68: AssertionError
...
```


### Information on your system

Running on LLNL's Quartz, where ``/usr/tce/packages/patchelf/patchelf-0.8/bin/patchelf`` symlinks ultimately to a bash script that uses `exec`.

--- a/lib/spack/spack/test/relocate.py
+++ b/lib/spack/spack/test/relocate.py
@@ -65,7 +65,7 @@ def test_file_is_relocatable(source_file, is_relocatable):
 )
 def test_patchelf_is_relocatable():
     patchelf = spack.relocate.get_patchelf()
-    assert spack.relocate.is_binary(patchelf)
+    assert llnl.util.filesystem.is_exe(patchelf)
     assert spack.relocate.file_is_relocatable(patchelf)
 
 

